apps/server/test/lexical/server/provider/code_action/replace_remote_function_test.exs

```diff
@@ -0,0 +1,144 @@
+defmodule Lexical.Server.Provider.CodeAction.ReplaceRemoteFunctionTest do
+  alias Lexical.Document
+  alias Lexical.Proto.Convert
+  alias Lexical.Protocol.Requests.CodeAction
+  alias Lexical.RemoteControl
+  alias Lexical.Server
+  alias Lexical.Server.Provider.Env
+  alias Lexical.Server.Provider.Handlers
+
+  import Lexical.Test.Protocol.Fixtures.LspProtocol
+  import Lexical.RemoteControl.Api.Messages
+  import Lexical.Test.Fixtures
+
+  use ExUnit.Case, async: false
+
+  setup_all do
+    start_supervised(Document.Store)
+    project = project(:code_actions)
+
+    {:ok, _} = start_supervised({DynamicSupervisor, Server.Project.Supervisor.options()})
+
+    {:ok, _} = start_supervised({Server.Project.Supervisor, project})
+
+    RemoteControl.Api.register_listener(project, self(), [project_compiled()])
+    RemoteControl.Api.schedule_compile(project, true)
+
+    assert_receive project_compiled(), 5000
+
+    path = file_path(project, Path.join("lib", "main.ex"))
+
+    {:ok, project: project, uri: Document.Path.ensure_uri(path)}
+  end
+
+  def build_request(path, line, char) do
+    uri = Document.Path.ensure_uri(path)
+
+    diagnostics = [
+      [
+        message: diagnostic_message(),
+        range: [end: [line: line + 1, character: 1], start: [line: line, character: char]]
+      ]
+    ]
+
+    params = [
+      context: [diagnostics: diagnostics],
+      range: [],
+      text_document: [uri: uri]
+    ]
+
+    with {:ok, _} <- Document.Store.open_temporary(uri),
+         {:ok, req} <- build(CodeAction, params) do
+      Convert.to_native(req)
+    end
+  end
+
+  defp diagnostic_message do
+    """
+    Remote.A.B.C.fun/0 is undefined or private. Did you mean:
+          * func/0
+    """
+  end
+
+  def handle(request, project) do
+    Handlers.CodeAction.handle(request, %Env{project: project})
+  end
+
+  test "replace remote function call", %{project: project, uri: uri} do
```

Comment:
A couple of things here. You're testing the integration of server and remote_control, but after the changes I made, all that's required of people implementing code actions is to just add a module in a list inside of remote_control. You made no changes to `server`. 

That said, integration tests are welcome. 

However, we've moved away from tests like these for the following reasons:
  1. They're very tedious to write
  2. It's not clear what they're testing
  3. It's not clear if the assertions themselves are correct.

If you wish to include these tests, you should extend `Lexical.Test.CodeMod.Case` and use the `apply_edits` function to ensure that the edits are what you intended. 

I'm also fine with you removing these tests and their supporting fixtures.

Reply:
They originally made more sense when I was creating a list of aliases that a function to replace could use. I wanted to test if that list of aliases was correct. However, now they seem to test `Lexical.Ast.expand_aliases` which is already tested in another place, so I am for deleting these tests.